devel/code/command.py

```diff
@@ -0,0 +1,231 @@
+
+        # create mount points definitions
+        for project_path in projects_to_mount:
+            # make sure that the project exists
+            if not os.path.isdir(project_path):
+                dtslogger.error(f"The path '{project_path}' is not a Duckietown project")
+            # get project info
+            proj = DTProject(project_path)
+            
+            root =  proj.path
+            # get local and remote paths to code
+            local_srcs, destination_srcs = proj.code_paths(root)
+            # compile mountpoints
+            for local_src, destination_src in zip(local_srcs, destination_srcs):
+                # Append to the list of mount points in DEVCONTAINER_TEMPLATE
+                DEVCONTAINER_TEMPLATE["mounts"].append(f"source={local_src},target={destination_src},type=bind")
+
+            # get local and remote paths to launchers
+            local_launchs, destination_launchs = proj.launch_paths(root)
+            if isinstance(local_launchs, str):
+                local_launchs = [local_launchs]
+                destination_launchs = [destination_launchs]
+            # compile mountpoints
+            for local_launch, destination_launch in zip(local_launchs, destination_launchs):
+                # Append to the list of mount points in DEVCONTAINER_TEMPLATE
+                DEVCONTAINER_TEMPLATE["mounts"].append(f"source={local_launch},target={destination_launch},type=bind")
+                # make sure the launchers are executable
+                try:
+                    _run_cmd(["chmod", "a+x", os.path.join(local_launch, "*")], shell=True)
+                except Exception:
+                    dtslogger.warning("An error occurred while making the launchers executable. "
+                                        "Things might not work as expected.")
+
+        # create image name
+        image = project.image(
+            arch=arch,
+            registry=registry_to_use,
+            owner=parsed.username,
+            version=project.distro,
+        )
+
+        DEVCONTAINER_TEMPLATE["image"] = image
+
+        # docker arguments
+        if not parsed.docker_args:
+            parsed.docker_args = []
+
+        # escape spaces in arguments
+        parsed.docker_args = [a.replace(" ", "\\ ") for a in parsed.docker_args]
+
+        DEVCONTAINER_TEMPLATE["runArgs"].extend(parsed.docker_args)
+
+        # Write DEVCONTAINER_TEMPLATE to .devcontainer/devcontainer.json
+        output_file_path = ".devcontainer/devcontainer.json"
+
+        # Create .devcontainer folder
+        if not os.path.exists('.devcontainer') and not os.path.isdir('.devcontainer'):
+            os.mkdir('.devcontainer')
+        with open(output_file_path, 'w') as json_file:
+            json.dump(DEVCONTAINER_TEMPLATE, json_file, indent=4,)
+
+        dtslogger.info(f'DEVCONTAINER_TEMPLATE written to {output_file_path}')
+        dtslogger.info("You can now open the devcontainer in your favorite editor")
```

Comment:
We can attempt to open VSCode in the project's location.

Reply:
Yes, there seems to be a shell api for devcontainers (other than the one already existing from VSCode) but wanted to keep this compatible with PyCharm as well. But we could look into opening VSCode already configured for easy development.

Reply:
Added.